URBNSwiftAlert/Classes/AlertStyler.swift

```diff
@@ -0,0 +1,448 @@
+//
+//  AlertStyler.swift
+//  Pods
+//
+//  Created by Kevin Taniguchi on 5/22/17.
+//
+//
+import Foundation
+import URBNSwiftyConvenience
+
+public struct AlertStyler {
+    
+    public var blur: Blur = Blur()
+    
+    public var background = Background()
+    
+    public var alertViewShadow = AlertViewShadow()
+    
+    public var animation = Animation()
+    
+    public var button = Button()
+    
+    public var cancelButton = CancelButton()
+    
+    public var destructiveButton = DestructiveButton()
+    
+    public var disabledButton = DisabledButton()
+    
+    public var title = Title()
+    
+    public var message = Message()
+    
+    public var textField = TextField()
+    
+    public var alert = Alert()
+    
+    /**
+     * The view you want to become the first responder when the alert view is finished presenting
+     * The alert position will adjust for the keyboard when using this property
+     */
+    public var firstResponder: UIView?
+}
+
+// MARK: Container Structs
+extension AlertStyler {
+    public struct Alert {
+        /**
+         * Min (greaterThanOrEqualTo) and Max widths (lessThanOrEqualTo) for the Alert -   The alert view will use its most compressed layout values for height
+         */
+        
+        public var minMaxWidth: (min: CGFloat, max: CGFloat)?
+        
+        /**
+         * Margins on left and right sides of the alert, which sets the width constraint.  Used if no min/max widths are given.  The alert view will use its most compressed layout values for height
+         */
+        public var horizontalMargin: CGFloat = 45.0 {
+            didSet {
+                assert(horizontalMargin > 0, "Margin must be greater than 0")
+            }
+        }
+        
+        /**
+         * Margin between sections in the alert. ie margin between the title and the message; message and the buttons, etc.
+         */
+        public var labelVerticalSpacing: CGFloat = 10.0
+        
+        /**
+         * Insets of the whole standard alert view when not using custom views.
+         */
+        public var insets = UIEdgeInsets(top: 24, left: 16, bottom: 5, right: 16)
+        
+        /**
+         * Corner radius of the alert view itself
+         */
+        public var cornerRadius: CGFloat = 0.0
+    }
+    
+    public struct Blur {
+        /**
+         * Pass no to disable blurring in the background
+         */
+        public var isEnabled = true
+    }
+    
+    public struct Background {
+        /**
+         * Background color of alert view
+         */
+        public var color = UIColor.white
+        /**
+         * Tint color of the view behind the Alert. Blur must be disabled
+         */
+        public var tint: UIColor?
```

Comment:
Is this the background of the AlertView itself? I‘m not sure `tint` makes sense within this type. I could see it being more related to the `Blur` type above _( 🤔 maybe as some sort of `enum`)_

Reply:
its not the color of the alert view. its the color of the alertviewcontroller.view, which is normally clear.  

it makes more sense to make it a part of the blur type.